stonybook/pipeline/quote/quote.py

```diff
@@ -0,0 +1,961 @@
+                            continue
+
+    def createConversationCsv(self, basepath, filename):
+        conversationQuotes = []
+        for convo in self.conversations:
+            quotesMerged = ""
+            for each in convo:
+                quotesMerged += f"{each.mention} : " + each.quote + "\n"
+            conversationQuotes.append({'quotes': quotesMerged})
+        df = pd.DataFrame(conversationQuotes)
+        df.to_csv(basepath + filename + '.csv')
+
+    def getConversations(self):
+        conversations = []
+        currentConversation = []
+
+        for i, quoteObj in enumerate(self.quotes):
+            if len(currentConversation) == 0:
+                currentConversation.append(quoteObj)
+            else:
+                previousSentenceId = int(currentConversation[-1].endSentId)
+                if int(quoteObj.startSentId) - previousSentenceId <= 3:
+                    currentConversation.append(quoteObj)
+                else:
+                    conversations.append(currentConversation)
+                    currentConversation = [quoteObj]
+
+        self.conversations = conversations
+
+    def searchUpwards(self, conversationIdx, currentIdx):
+        i = currentIdx - 2
+        while (i >= 0):
+            if self.conversations[conversationIdx][i].mention != None:
+                self.conversations[conversationIdx][currentIdx].mention = self.conversations[conversationIdx][i].mention
+                self.conversations[conversationIdx][currentIdx].mention_start = self.conversations[conversationIdx][
+                    i].mention_start
+                self.conversations[conversationIdx][currentIdx].mention_end = self.conversations[conversationIdx][
+                    i].mention_end
+                return True
+            i = i - 2
+        return False
+
+    def searchDownwards(self, conversationIdx, currentIdx):
+        i = currentIdx + 2
+        while (i < len(self.conversations[conversationIdx])):
+            if self.conversations[conversationIdx][i].mention != None:
+                self.conversations[conversationIdx][currentIdx].mention = self.conversations[conversationIdx][i].mention
+                self.conversations[conversationIdx][currentIdx].mention_start = self.conversations[conversationIdx][
+                    i].mention_start
+                self.conversations[conversationIdx][currentIdx].mention_end = self.conversations[conversationIdx][
+                    i].mention_end
+                return True
+            i = i + 2
+        return False
+
+    def correctMentionsInConversations(self):
+        for i, conversation in enumerate(self.conversations):
+            for j, quoteObj in enumerate(conversation):
+                if quoteObj.mention == None:
+                    status = self.searchUpwards(i, j)
+                    if status == False:
+                        self.searchDownwards(i, j)
+        self.updateQuotes()
```

Comment:
Our understanding is that the quote objects in your self.conversations member variable are the same quote objects by reference as the ones in the self.quotes member variable. Thus, updateQuotes does not seem necessary. Can you verify this to be true through some assertion or if-check?